### PR TITLE
test: Upgrade integration test from 1.7 to master

### PR DESCRIPTION
**Pull request by travisn**
**Description of your changes:**
The upgrade integration test was from rook v1.6 to the latest master. This was necessary until we are ready for the v1.8 release, from which time we want to focus the upgrade testing from v1.7 to the latest master.

The duplication in the test CRs and other resources is now reduced by the upgrade calling a thin wrapper to forward a call to the master version of the resource. When a new feature is added that needs to be differentiated from the previous version, the method then can be implemented instead of wrapping the master implementation.

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

File: tests/framework/installer/ceph_manifests_previous.go [hunk 1]
```diff
@@ -0,0 +1,162 @@
+/*
+Copyright 2016 The Rook Authors. All rights reserved.
```
Comment (leseb) [inline tests/framework/installer/ceph_manifests_previous.go:2]:
```suggestion
Copyright 2021 The Rook Authors. All rights reserved.
```

File: tests/framework/installer/ceph_manifests_previous.go [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package installer
+
+import (
+	"strings"
+
```
Comment (leseb) [inline tests/framework/installer/ceph_manifests_previous.go:21]: blank line?

**Review by leseb — CHANGES_REQUESTED** [review]
Just nits

File: tests/framework/installer/ceph_manifests_previous.go (hunk 2 above)
Comment (travisn) [reply tests/framework/installer/ceph_manifests_previous.go:21]: gofmt is automatically adding it in, to separate the built-in import from the fully qualified import

**Pull request merged**